R/summary_funs.R

```diff
@@ -406,92 +438,120 @@ get_stat <- function(mu, lppd, y_wobs_test, stat, mu.bs = NULL, lppd.bs = NULL,
-      # be adapted:
-      n_notna <- sum(!is.na(mu))
-      if (!is.null(n_notna.bs) &&
-          getOption("projpred.additional_checks", FALSE)) {
-        stopifnot(n_notna == n_notna.bs)
+      if (!is.null(summaries_baseline)) {
+        mu_baseline <- rep(summaries_baseline$mu, y_wobs_test$wobs)
+      } else {
+        mu_baseline <- NULL
       }
       wcv <- rep(wcv, y_wobs_test$wobs)
-      wcv <- n_notna * wcv / sum(wcv)
+      wcv <- n * wcv / sum(wcv)
     } else {
       stopifnot(all(y_wobs_test$wobs == 1))
+      if (!is.null(summaries_baseline)) {
+        mu_baseline <- summaries_baseline$mu
+      } else {
+        mu_baseline <- NULL
+      }
     }
     if (stat %in% c("acc", "pctcorr")) {
       # Find out whether each observation was classified correctly or not:
       if (!is.factor(mu)) {
         mu <- round(mu)
       }
-      crrct <- mu == y
+      correct <- mu == y
 
-      if (!is.null(mu.bs)) {
-        if (!is.factor(mu.bs)) {
-          mu.bs <- round(mu.bs)
+      if (!is.null(mu_baseline)) {
+        if (!is.factor(mu_baseline)) {
+          mu_baseline <- round(mu_baseline)
         }
-        crrct.bs <- mu.bs == y
+        correct_baseline <- mu_baseline == y
+      } else {
+        correct_baseline <- 0
+      }
 
-        value <- mean(wcv * (crrct - crrct.bs), na.rm = TRUE)
-        value.se <- weighted.sd(crrct - crrct.bs, wcv, na.rm = TRUE) /
-          sqrt(n_notna)
+      if (!is.null(summaries_fast) && sum(n_loo<n)) {
+        # subsampling difference estimator (Magnusson et al., 2020)
+        mu_fast <- summaries_fast$mu
+        if (!is.factor(mu_fast)) {
+          mu_fast <- round(mu_fast)
+        }
+        correct_fast <- mu_fast == y
+        srs_diffe <- .srs_diff_est_w(y_approx = correct_fast - correct_baseline,
+                                     y = (correct-correct_baseline)[loo_inds],
+                                     y_idx = loo_inds,
+                                     w = wcv)
+        value <- srs_diffe$y_hat / n + mean(wcv * correct_baseline)
+        # combine estimates of var(y_hat) and var(y)
+        value_se <- sqrt(srs_diffe$v_y_hat + srs_diffe$hat_v_y) / n
       } else {
-        value <- mean(wcv * crrct, na.rm = TRUE)
-        value.se <- weighted.sd(crrct, wcv, na.rm = TRUE) / sqrt(n_notna)
+        # full LOO estimator
+        value <- mean(wcv * correct)
+        value_se <- .weighted_sd(correct - correct_baseline, wcv) / sqrt(n)
       }
     } else if (stat == "auc") {
-      if (!is.null(mu.bs)) {
-        auc.data <- cbind(y, mu, wcv)
-        auc.data.bs <- cbind(y, mu.bs, wcv)
-        value <- auc(auc.data) - auc(auc.data.bs)
-        idxs_cols <- seq_len(ncol(auc.data))
-        idxs_cols_bs <- setdiff(seq_len(ncol(auc.data) + ncol(auc.data.bs)),
+      if (!is.null(summaries_fast) && sum(n_loo<n)) {
+        # subsampling LOO with AUC not implemented. Using fast LOO result.
+        mu <- summaries_fast$mu
```

Comment:
Is `summaries_fast$mu` guaranteed to have no `NA`s? I'm asking because above, `wcv <- n_notna * wcv / sum(wcv)` was replaced by `wcv <- n * wcv / sum(wcv)` and `wcv` is used shortly after this line.

Reply:
I think if there are NAs, they should not be ignored and the result should be NA, too (or an error)

Reply:
Ok, I see. Then this should be fine.